packages/schematics/angular/application/index.ts

```diff
@@ -21,89 +23,113 @@ import {
   template,
   url,
 } from '@angular-devkit/schematics';
-import {
-  NodePackageInstallTask,
-  NodePackageLinkTask,
-  RepositoryInitializerTask,
-} from '@angular-devkit/schematics/tasks';
 import { Schema as ApplicationOptions } from './schema';
 
 
-function minimalPathFilter(path: string): boolean {
-  const toRemoveList: RegExp[] = [/e2e\//, /editorconfig/, /README/, /karma.conf.js/,
-                                  /protractor.conf.js/, /test.ts/, /tsconfig.spec.json/,
-                                  /tslint.json/, /favicon.ico/];
-
-  return !toRemoveList.some(re => re.test(path));
+function updateWorkspaceFile(options: ApplicationOptions, newProjectRoot: string): Rule {
+  return (host: Tree, context: SchematicContext) => {
+    context.logger.info(`Updating workspace file`);
+    const workspacePath = '/angular.json';
+    const workspaceBuffer = host.read(workspacePath);
+    if (workspaceBuffer === null) {
+      throw new SchematicsException(`Configuration file (${workspacePath}) not found.`);
+    }
+    const workspaceAst = parseJsonAst(workspaceBuffer.toString());
+    if (workspaceAst.value === null) {
+      throw new SchematicsException(`Unable to parse configuration file (${workspacePath}).`);
+    }
+    // tslint:disable-next-line:no-any
+    const projects: JsonObject = (<any> workspaceAst.value).projects || {};
+    const rootPath = `${newProjectRoot}/${options.name}`;
+    const parentDirMap = rootPath.split('').filter(x => x === '/').map(_ => '../');
+    const project = {
+      root: rootPath,
+      projectType: 'application',
+      cli: {},
+      schematics: {},
+      architect: {
+        targets: {
+          build: {
+            builder: '@angular-devkit/build-webpack:browser',
+            transforms: [
+              {
+                plugin: '@angular-devkit/architect-transforms:replacement',
+                file: 'environments/environment.ts',
+                configurations: {
+                  production: 'environments/environment.prod.ts',
+                },
+              },
+            ],
+            options: {
+              outputPath: `${parentDirMap}/dist`,
+              index: 'index.html',
```

Comment:
All file paths are root relative, so these need to be prefixed by `rootPath`. Have a look at the existing `__dot__angular.json` if you're unsure which need the root dir, it currently has `<%= sourceDir %>` where needed.